### PR TITLE
New downloading features along with adaptations to the new APIs

### PR DESCRIPTION
**Pull request by dzhuang**
## Proposed changes
- Enable multiple subtitles and transcripts with alternatives.
Sometimes we need to download subititles in different languages, for example, 'en' and 'zh-CN', which is not supported currently. The PR enabled downloading multiple subtitles & transcripts, along with fallback for each language, by using the following format:
```
--subtitle-language <lang1,lang2|fallback1_for_lang2|fallback2_for_lang2>
```
This can will also be an alternative fix for #596 .
- Enable downloading resources blocks.
Some materials are now in a new blocks named ``Resources``, which corresponed to the new API ``onDemandReferences.v1``. Fixing #597 

- Enable file to store parameters.  
There are cases where there are too many parameters (besides username and password) to be set each time we download a course. I use ``configargparse`` to read configurations stored in a local file named ``coursera-dl.conf``. Fixing #97 

- Enable downloading new typename ``programming``
That is different from original ``gradedProgramming`` and ``ungradedProgramming`` so it is not recognized by coursera-dl currently. An example can be find at the end of [this page].
The new API is ``onDemandProgrammingImmediateInstructions.v1``.

- Allow alternative ``MathJax.js`` cdn url when downloading webpages, as the default [``MathJax.js`` url](https://cdn.mathjax.org/mathjax/latest/MathJax.js) is not accessible in some countries (this is the case at least for me).

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](/CONTRIBUTING.md) doc
- [x] I agree to contribute my changes under the project's [LICENSE](/LICENSE)
- [x] I have checked that the unit tests pass locally with my changes
- [x] I have checked the style of the new code (lint/pep).
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

Sorry for putting these features altogether, I should have submit them serperately.

### Reviewers
@rbrito @balta2ar Thanks in advance.

### CODE REVIEW

File: coursera/extractors.py [hunk 1]
```diff
@@ -174,4 +188,41 @@ def _parse_on_demand_syllabus(self, page, reverse=False,
         if modules and reverse:
             modules.reverse()
 
+        course_id = dom['id']
+        try:
+            references_poll= self._get_references_poll(course_id)
+        except:
```
Comment (balta2ar) [inline coursera/extractors.py]: This is too broad. Please be more precise about what exception you are expecting here. Network error, parsing error?

File: coursera/extractors.py [hunk 2]
```diff
@@ -174,4 +188,41 @@ def _parse_on_demand_syllabus(self, page, reverse=False,
...
+
+        references = None
+        if references_poll_dom:
+            references = references_poll_dom['elements']
+
+        reference_modules = []
+        if references:
+            logging.info('Processing resources')
+            for reference in references:
+                ref_dl = []
```
Comment (balta2ar) [inline coursera/extractors.py]: What is `ref_dl`? Please use more descriptive name. `reference_download_links`?

**Comment by dzhuang** [comment]
@balta2ar Thanks for your advise. I've modified the code and added unit-test, and all tests had passed.

**Comment by balta2ar** [comment]
Thank you for the contribution!

**Pull request merged**